pkg/tfbridge/provider.go

```diff
@@ -59,6 +59,63 @@ type Resource struct {
 	TFName string           // the Terraform resource name.
 }
 
+// collectImporterAttributes runs the Terraform Importer defined on the Resource for the given
+// resource ID, and adds any non-ID attributes to the inputs map.
+func (res *Resource) collectImporterAttributes(resourceID resource.ID, inputs map[string]string) error {
+	// There is nothing to do here if the resource doesn't have an importer defined in the
+	// Terraform schema.
+	if res.TF.Importer == nil {
+		return nil
+	}
+
+	id := resourceID.String()
+
+	glog.V(9).Infof("%s has TF Importer", res.TFName)
+
+	// Prepare a Terraform ResourceData for the importer
+	data := res.TF.Data(nil)
+	data.SetId(id)
+	data.SetType(res.TFName)
+
+	results, err := res.TF.Importer.State(data, nil)
+	if err != nil {
+		return errors.Wrapf(err, "importing %s", id)
+	}
+
+	if len(results) < 1 {
+		return errors.Errorf("importer for %s returned no resources", id)
+	}
+
+	// Find the resource with the matching ID, since an importer can return several (for example,
+	// a security group and associated rules), and set any non-id attributes as inputs to read.
+	for _, result := range results {
+		if result.Id() != id {
```

Comment:
We'll want to check the type as well, right?

Reply:
Good catch, that needs adding too.

Reply:
I've added a check for this after we verify that State is not nil, in b2b9be9.

Reply:
It might be a good idea to restructure this code to separate out the search for the correct state from the action performed on it. I'll push another commit shortly.